controllers/hostedcontrolplane/hostedcontrolplane.go

```diff
@@ -323,3 +458,277 @@ func (r *HostedControlPlaneReconciler) SetupWithManager(mgr ctrl.Manager) error
+func (APIClient *APIClient) makeRequest(method, path string, renderedJSON string) (*http.Response, error) {
+	url := APIClient.baseURL + path
+	var reqBody io.Reader
+	if renderedJSON != "" {
+		reqBody = bytes.NewBufferString(renderedJSON)
+	}
+
+	req, err := http.NewRequest(method, url, reqBody)
+	if err != nil {
+		return nil, err
+	}
+
+	req.Header.Set("Authorization", "Api-Token "+APIClient.apiToken)
+	req.Header.Set("Content-Type", "application/json")
+
+	return APIClient.httpClient.Do(req)
+}
+
+func (r *HostedControlPlaneReconciler) getSecret(ctx context.Context) (string, string, error) {
+
+	secret := &v1.Secret{}
+	err := r.Client.Get(ctx, types.NamespacedName{Name: secretName, Namespace: secretNamespace}, secret)
+	if err != nil {
+		return "", "", fmt.Errorf("error getting Kubernetes secret: %v", err)
+	}
+
+	valueBytesDynatraceApiToken, ok := secret.Data[dynatraceApiKey]
+	if !ok {
+		return "", "", fmt.Errorf("secret did not contain key %s", dynatraceApiKey)
+	}
+	if len(valueBytesDynatraceApiToken) == 0 {
+		return "", "", fmt.Errorf("%s is empty", dynatraceApiKey)
+	}
+	valueDynatraceApiToken := string(valueBytesDynatraceApiToken)
+
+	valueBytesDynatraceTenant, ok := secret.Data[dynatraceTenantKey]
+	if !ok {
+		return "", "", fmt.Errorf("secret did not contain key %s", dynatraceTenantKey)
+	}
+	if len(valueBytesDynatraceTenant) == 0 {
+		return "", "", fmt.Errorf("%s is empty", dynatraceTenantKey)
+	}
+	valueDynatraceTenant := string(valueBytesDynatraceTenant)
+
+	return valueDynatraceApiToken, valueDynatraceTenant, nil
+}
+
+func (APIClient *APIClient) getDynatraceHTTPMonitorID(ctx context.Context, log logr.Logger, hostedcontrolplane *hypershiftv1beta1.HostedControlPlane) (string, error) {
+	labels := hostedcontrolplane.GetLabels()
+	dynatraceHttpMonitorId, exists := labels[httpMonitorLabel]
+	if exists {
+		return dynatraceHttpMonitorId, nil
+	}
+	return "", fmt.Errorf("key '%s' not found in labels", httpMonitorLabel)
+}
+
+func (r *HostedControlPlaneReconciler) UpdateHostedControlPlaneLabels(ctx context.Context, hostedcontrolplane *hypershiftv1beta1.HostedControlPlane, key, value string) error {
+	labels := hostedcontrolplane.GetLabels()
+	labels[key] = value
+	hostedcontrolplane.SetLabels(labels)
+
+	err := r.Client.Update(ctx, hostedcontrolplane)
+	if err != nil {
+		return fmt.Errorf("error updating hostedcontrolplane monitor: %v", err)
+	}
+	return nil
+}
+
+func (r *HostedControlPlaneReconciler) GetAPIServerHostname(hostedcontrolplane *hypershiftv1beta1.HostedControlPlane, log logr.Logger) (string, error) {
+	for _, service := range hostedcontrolplane.Spec.Services {
+		if service.Service == "APIServer" {
+			return service.ServicePublishingStrategy.Route.Hostname, nil
+		}
+	}
+	return "", fmt.Errorf("APIServer service not found in the hostedcontrolplane")
+}
+
+func (APIClient *APIClient) getDynatraceEquivalentClusterRegionId(hostedcontrolplane *hypershiftv1beta1.HostedControlPlane) (string, error) {
```

Comment:
I think it might be better to try to keep `hostedcontrolplane` objects as decoupled from `dynatrace` objects as much as possible. In this function I see that we're only using the `hostedcontrolplane.Spec.Platform.AWS.Region` out of the object. How would you feel about just taking a string as an argument and letting the caller of this function handle figuring out how to get the string it needs.  This will result in less-coupled code but also would make this easier to test.